iceoryx_hoofs/test/moduletests/test_cxx_functional_interface_and_then.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+#include "test_cxx_functional_interface_types.hpp"
+
+namespace
+{
+using namespace test_cxx_functional_interface;
+using namespace ::testing;
+
+#define IOX_TEST(TestName, variationPoint)                                                                             \
+    using SutType = typename TestFixture::TestFactoryType::Type;                                                       \
+    TestName<iox::cxx::internal::HasValueMethod<SutType>::value>::template performTest<                                \
```

Comment:
To make this a bit more readable, What do you think about the following?
```suggestion
    constexpr auto HAS_VALUE_METHOD = iox::cxx::internal::HasValueMethod<SutType>::value; \
    TestName<HAS_VALUE_METHOD>::template performTest<                                                       \
```
Maybe with a better name for the constexpr?

It would reduce the cognitive load when parsing in that line